api/config.yaml

```diff
@@ -13,6 +13,8 @@ components:
     ConnectionResult:
       type: object
       properties:
+        endpointLinting:
+          $ref: "#/components/schemas/ConnectionTestStep"
```

Comment:
I'm not a huge fan of having these be strict entries of the connection result object, I think this should be an array with different shapes of objects it can return, maybe having a type.

Not something for this PR tho 😄